embark-consult.el

```diff
@@ -111,88 +103,83 @@ If set to nil `occur-mode' is used."
-      (funcall resutls-writer lines)
+      (funcall results-writer)
       (goto-char (point-min))
       (occur-mode))
     (pop-to-buffer buf)))
 
-(defun embark-consult-export-location (lines)
+(defun embark-consult-export-location-occur (lines)
+  "Create an occur or grep mode buffer listing LINES.
+The elements of LINES are assumed to be values of category `consult-location'."
+  (funcall
+   #'embark-consult--setup-occur-mode
+   (lambda ()
+     "Format the result lines from `consult-location' range of commands
+for the occur-mode.
+You can edit results in place with `occur-edit-mode'"
+     (let ((mouse-msg "mouse-2: go to this occurrence")
+            last-buf)
+        (dolist (line lines)
+          (pcase-let*
+              ((`(,loc . ,num) (consult--get-location line))
+               ;; the text properties added to the following strings are
+               ;; taken from occur-engine
+               (lineno (propertize (format "%7d:" num)
+                                   'occur-prefix t
+                                   ;; Allow insertion of text at the end
+                                   ;; of the prefix (for Occur Edit mode).
+                                   'front-sticky t
+                                   'rear-nonsticky t
+                                   'occur-target loc
+                                   'follow-link t
+                                   'help-echo mouse-msg))
+               (contents (propertize (embark-consult--strip line)
+                                     'occur-target loc
+                                     'occur-match t
+                                     'follow-link t
+                                     'help-echo mouse-msg))
+               (nl (propertize "\n" 'occur-target loc))
+               (this-buf (marker-buffer loc)))
+            (unless (eq this-buf last-buf)
+              (insert (propertize
+                       (format "lines from buffer: %s\n" this-buf)
+                       'face list-matching-lines-buffer-name-face))
+              (setq last-buf this-buf))
+            (insert (concat lineno contents nl))))))))
+
+(defun embark-consult-export-location-grep (lines)
   "Create an occur or grep mode buffer listing LINES.
 The elements of LINES are assumed to be values of category `consult-location'."
-  (if embark-consult-export-location-candidates-to-grep-mode
-      (funcall #'embark-consult--setup-grep-mode
-               "Exported Consult results\n\n"
-               lines
-               #'embark-consult--location-to-grep-results-writer)
-    (funcall #'embark-consult--setup-occur-mode
-             lines
-             #'embark-consult--location-to-occur-results-writer)))
+  (funcall
+   #'embark-consult--setup-grep-mode
+    "Exported line search results (file-backed buffers only):.\n\n"
+   (lambda ()
+      "Format the results from `consult-location' range of commands for the `grep-mode' as alternative to occur.
```

Comment:
Is there any way the user can see these docstring for inner lambdas? I think there isn't, and if I'm right then these docstrings are only for readers of the source code, in which case they would be better off as comments.